cabal-install/Distribution/Client/ProjectOrchestration.hs

```diff
@@ -342,16 +365,18 @@ resolveAndCheckTargets targetDefaultComponents
       = Left (BuildTargetNotInProject (buildTargetPackage t))
 
 
-    projAllPkgs, projLocalPkgs :: Map PackageName InstalledPackageId
+    -- NB: Make sure you get the *un-munged* PackageName
+    projAllPkgs, projLocalPkgs :: Map PackageName [InstalledPackageId]
```

Comment:
It's introduced by this patch. We're creating a mapping from *un*-munged package name to installed package id for every *component* in the install plan; every component from a given package has the same package name so there are duplicates. We need all of them, because if a user asks for some package we need to make sure we look at all the components for that package.